detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/BitbakeExtractor.java

```diff
@@ -73,34 +72,27 @@ public Extraction extract(
         ExecutableTarget bash
     ) {
         List<CodeLocation> codeLocations = new ArrayList<>();
-
         BitbakeSession bitbakeSession = new BitbakeSession(executableRunner, bitbakeRecipesParser, sourceDirectory, buildEnvScript, sourceArguments, bash, toolVersionLogger, buildFileFinder, bitbakeEnvironmentParser);
         bitbakeSession.logBitbakeVersion();
         File buildDir = bitbakeSession.determineBuildDir();
         BitbakeEnvironment bitbakeEnvironment = bitbakeSession.executeBitbakeForEnvironment();
-        for (String packageName : packageNames) {
-            Map<String, String> imageRecipes = null;
+        ShowRecipesResults showRecipesResults;
+        try {
+            showRecipesResults = bitbakeSession.executeBitbakeForRecipeLayerCatalog();
+        } catch (IOException | ExecutableFailedException e) {
+            String msg = String.format("Error collecting recipe layer information from show-recipes command: %s", e.getMessage());
+            logger.error(msg);
+            return new Extraction.Builder().failure(msg).build();
+        }
+        for (String targetImage : packageNames) {
             try {
-                if (dependencyTypeFilter.shouldExclude(BitbakeDependencyType.BUILD)) {
-                    imageRecipes = readImageRecipes(buildDir, packageName, bitbakeEnvironment, followSymLinks, searchDepth);
-                }
-                BitbakeGraph bitbakeGraph = generateBitbakeGraph(bitbakeSession, buildDir, packageName, followSymLinks, searchDepth);
-                List<BitbakeRecipe> bitbakeRecipes = bitbakeSession.executeBitbakeForRecipeLayerCatalog();
-                Map<String, String> recipeNameToLayersMap = bitbakeRecipesToLayerMap.convert(bitbakeRecipes);
-
-                DependencyGraph dependencyGraph = bitbakeGraphTransformer.transform(bitbakeGraph, recipeNameToLayersMap, imageRecipes);
-                CodeLocation codeLocation = new CodeLocation(dependencyGraph);
-
-                codeLocations.add(codeLocation);
-
+                codeLocations.add(generateCodeLocationForTargetImage(followSymLinks, searchDepth, dependencyTypeFilter, bitbakeSession, buildDir, bitbakeEnvironment, showRecipesResults, targetImage));
             } catch (IOException | IntegrationException | NotImplementedException | ExecutableFailedException e) {
-                logger.error(String.format("Failed to extract a Code Location while running Bitbake against package '%s': %s", packageName, e.getMessage()));
+                logger.error(String.format("Failed to extract a Code Location while running Bitbake against package '%s': %s", targetImage, e.getMessage()));
```

Comment:
Same thing, you could just bubble these up.

Reply:
Here we want to just log and continue on to the next target image / codelocation